templates/discover/lessons.html.twig

```diff
@@ -0,0 +1,42 @@
+{#
+ # ---------------------------------------------------------------------
+ #
+ # GLPI - Gestionnaire Libre de Parc Informatique
+ #
+ # http://glpi-project.org
+ #
+ # @copyright 2015-2023 Teclib' and contributors.
+ # @copyright 2003-2014 by the INDEPNET Development Team.
+ # @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ #
+ # ---------------------------------------------------------------------
+ #
+ # LICENSE
+ #
+ # This file is part of GLPI.
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+<div class="table-responsive">
+    <table class="table card-table table-vcenter">
+        <tbody>
+            {% for lesson in lessons %}
+                {% include 'discover/lesson.html.twig' with {'progression': progressions[lesson['id']]} %}
```

Comment:
We are trying to avoid sharing the whole context to twig templates because it gets hard to maintain over time.
It means you should use the `with_context = false` param and share explicitly the `lesson` object.

Also, something I've learned recently is that we should avoid the `include` tag and use the `include` _function_ instead (as described here: https://twig.symfony.com/doc/3.x/tags/include.html).